transcription/utils.go

```diff
@@ -86,6 +86,73 @@ func filePathFromURL(url string) string {
 	return filePath
 }
 
+// SplitFlacFile ensures that the input audio files to IBM are less than 100mb, with 5 seconds of redundancy between files.
+func SplitFlacFile(fn string) error {
+	// http://stackoverflow.com/questions/36632511/split-audio-file-into-several-files-each-below-a-size-threshold
+	// The Stack Overflow answer ultimately calculated the length of each audio chunk in seconds.
+	// chunk_length_in_sec = math.ceil((duration_in_sec * file_split_size ) / wav_file_size)
+	// Invariant: If ConvertAudioIntoWavFormat is called on fn, a 95MB chunk of resulting Wav file is always 2968 seconds.
+	// In the above equation, there is one constant: file_split_size = 95000000 bytes.
+	// duration_in_sec is used to calculate wav_file_size, so it is canceled out in the ratio.
+	// wav_file_size = (sample_rate * bit_rate * channel_count * duration_in_sec) / 8
+	// sample_rate = 44100, bit_rate = 16, channels_count = 1 (stereo: 2, but Sphinx prefers 1)
+	// As a chunk of the Wav file is extracted using FFMPEG, it is converted back into Flac format.
+	wavFileName, err := ConvertAudioIntoFormat(fn, "wav")
+	if err != nil {
+		return err
+	}
+
+	numChunks, err := getNumChunks(fn)
+	if err != nil {
+		return err
+	}
+
+	chunkLengthInSeconds := 2968
+	for i := 0; i < numChunks; i++ {
+		// 5 seconds of redundancy for each chunk after the first
+		startingSecond := i*chunkLengthInSeconds - (i-1)*5
+		newFileName := wavFileName + strconv.Itoa(i)
+		if err := extractAudioSegment(newFileName, startingSecond, chunkLengthInSeconds); err != nil {
```

Comment:
how will the caller know what the produced files are? should we return a slice of strings indicating the file paths in order?